### PR TITLE
Features: Implemented ChannelFirst2d

### PR DESCRIPTION
**Pull request by Pwhsky**
This PR implements ChannelFirst2d extended to handle torch tensors, as well as adding unit tests for torch tensors.

### CODE REVIEW

File: deeptrack/features.py [hunk 1]
```diff
@@ -6192,37 +6194,45 @@ class ChannelFirst2d(Feature):
 
     def __init__(
         self: Feature,
-        axis: int = -1,
+        axis: PropertyLike[int] = -1,
```
Comment (mirjagranfors) [inline deeptrack/features.py:6902]: Should axis be PropertyLike[int]? I don't really see why it would ever be something other than a fixed integer.

File: deeptrack/features.py [hunk 2]
```diff
@@ -6192,37 +6194,45 @@ class ChannelFirst2d(Feature):
...
+        **kwargs: dict of str to Any
             Additional keyword arguments passed to the parent `Feature` class.
 
         """
+        import warnings
+
+        warnings.warn(
+            "ChannelFirst2d is deprecated and may be removed in a future release. "
+            "The current implementation is not guaranteed to be exactly "
+            "equivalent to prior implementations. "
```
Comment (mirjagranfors) [inline deeptrack/features.py]: I think that a comma is missing after the string. Without it I get an error

File: deeptrack/features.py [hunk 3]
```diff
@@ -6147,19 +6147,21 @@ class ChannelFirst2d(Feature):
 
     This feature rearranges the axes of a 3D image so that the specified axis
```
Comment (mirjagranfors) [inline deeptrack/features.py]: Some lines have blank spaces at the end. Check it and remove the spaces.

File: deeptrack/features.py (hunk 2 above)
Comment (mirjagranfors) [reply deeptrack/features.py]: Also, a line goes beyond 79 characters

File: deeptrack/features.py [hunk 4]
```diff
@@ -6192,37 +6194,45 @@ class ChannelFirst2d(Feature):
...
+            DeprecationWarning,
+        )
 
         super().__init__(axis=axis, **kwargs)
 
     def get(
         self: Feature,
-        image: np.ndarray,
+        image: NDArray | torch.Tensor,
         axis: int,
```
Comment (mirjagranfors) [inline deeptrack/features.py]: If "axis = -1" in the init, shouldn't it also be so here in the "get"? Or remove it in both cases.

File: deeptrack/tests/test_features.py [hunk 5]
```diff
@@ -1957,6 +1957,16 @@ def test_ChannelFirst2d(self):
...
         self.assertEqual(output_image.shape, (3, 10, 20))
 
+        if TORCH_AVAILABLE:
+            input_image_2d = torch.rand(10, 20)
+            output_image = channel_first_feature.get(input_image_2d, axis=-1)
+            self.assertEqual(tuple(output_image.shape), (1, 10, 20))
+
+            input_image_3d = torch.rand(10, 20, 3)
+            output_image_3d = channel_first_feature.get(input_image_3d,
+                                                        axis=-1)
```
Comment (mirjagranfors) [inline deeptrack/tests/test_features.py]: Does this work for you? When I try to run this in my script I get an error when using a 3d torch tensor and axis=-1

File: deeptrack/tests/test_features.py [hunk 6]
```diff
@@ -1957,6 +1957,16 @@ def test_ChannelFirst2d(self):
...
+            self.assertEqual(tuple(output_image_3d.shape), (3, 10, 20))
+
```
Comment (mirjagranfors) [inline deeptrack/tests/test_features.py:2003]: I think that it would be nice to add some test with the actual values of the arrays and tensors. They can of course be small (e.g. [[1, 2, 3], [4, 5, 6]])

File: deeptrack/features.py [hunk 7]
```diff
@@ -6287,7 +6305,7 @@ class Upscale(Feature):
 
     Methods
```
Comment (mirjagranfors) [inline deeptrack/features.py:7018]: I think that it becomes easier to review if you only put one class in each PR, so a separate one for the "Upscale" would have been nicer

File: deeptrack/features.py [hunk 8]
```diff
@@ -6287,7 +6305,7 @@ class Upscale(Feature):
...
     -------
-    `get(image: np.ndarray | Image, factor: int | tuple[int, int, int], **kwargs) -> np.ndarray`
+    `get(image: np.ndarray | Image | torch.tensor, factor: int | tuple[int, int, int], **kwargs) -> np.ndarray | torch.tensor`
```
Comment (mirjagranfors) [inline deeptrack/features.py:7020]: This line, and also some other line, is longer than 79 characters

File: deeptrack/features.py [hunk 9]
```diff
@@ -6192,37 +6194,45 @@ class ChannelFirst2d(Feature):
...
 
         Rearrange the axes of a 3D image to channel-first format or add a 
         channel dimension to a 2D image.
 
         Parameters
         ----------
-        image: np.ndarray
+        image: NDArray | torch.Tensor 
             The input image to process. Can be 2D or 3D.
         axis: int
```
Comment (mirjagranfors) [inline deeptrack/features.py:6942]: Maybe add that "axis" doesn't affect anything for 2D images

File: deeptrack/features.py [hunk 10]
```diff
@@ -6364,15 +6382,15 @@ def __init__(
 
     def get(
         self: Feature,
-        image: np.ndarray,
+        image: np.ndarray | torch.tensor,
         factor: int | tuple[int, int, int],
```
Comment (mirjagranfors) [inline deeptrack/features.py:7098]: If there is the default in init, schouldn't we have the same default value here?

File: deeptrack/features.py (hunk 7 above)
Comment (mirjagranfors) [reply deeptrack/features.py:7018]: To me it doesn't seem like "Upscale" is ready yet. Let me know when you want a review on it

File: deeptrack/tests/test_features.py (hunk 5 above)
Comment (Pwhsky) [reply deeptrack/tests/test_features.py]: Thank you for pointing this out, I have added a fix for this that passes the tests properly.
